source/include/FreeRTOS_Sockets.h

```diff
@@ -367,6 +367,13 @@
 /* For internal use only: return the connection status. */
         BaseType_t FreeRTOS_connstatus( ConstSocket_t xSocket );
 
+/* For advanced applications only:
+ * Get a direct pointer to the beginning of the circular transmit buffer.
+ * In case the buffer was not yet created, it will be created in
+ * this call.
+ */
+        uint8_t * FreeRTOS_get_tx_base( Socket_t xSocket );
```

Comment:
Should we wrap this with ipconfigUSE_TCP?

Reply:
It is already conditional on `ipconfigUSE_TCP`:
~~~
    #if ( ipconfigUSE_TCP == 1 )           // line 270
        #if ( ipconfigUSE_TCP == 1 )       // line 238
            BaseType_t FreeRTOS_ReleaseTCPPayloadBuffer( Socket_t xSocket,
                                                         void const * pvBuffer,
                                                         BaseType_t xByteCount );
        #endif /* ( ipconfigUSE_TCP == 1 ) */  // line 336
        uint8_t * FreeRTOS_get_tx_base( Socket_t xSocket );  // line 372
    #endif /* ( ipconfigUSE_TCP == 1 ) */      // line 393
~~~

I will remove the nested `#if/#endif` couple.